pkg/manifest/manifest.go

```diff
@@ -0,0 +1,118 @@
+package manifest
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// Manifest stores Kubernetes object in Raw from a file.
+// It stores the GroupVersionKind for the manifest.
+type Manifest struct {
+	// OriginalFilename is set to the filename this manifest was loaded from.
+	// It is not guaranteed to be set or be unique, but we will set it when
+	// loading from disk to provide better debuggability.
+	OriginalFilename string
+
+	Raw []byte
+	GVK schema.GroupVersionKind
+
+	Obj *unstructured.Unstructured
+}
+
+// UnmarshalJSON unmarshals bytes of single kubernetes object to Manifest.
+func (m *Manifest) UnmarshalJSON(in []byte) error {
```

Comment:
why is this no constructor?

Reply:
Not sure as this is existing code I simply migrated "as is" from CVO. The guidance I was given is to effect as little change as possible to the CVO code. I can add a constructor but forcing it's use by making the Manifest struct private would have big impact to CVO. Would providing a constructor but leaving Manifest public suffice?

Reply:
added a ctor

Reply:
why not make this func private?

Reply:
The answer to my questions: `UnmarshalJSON` is called by the yaml or json decoder. In other words, `Manifest` knows how to unmarshal itself. This means we must have this signature. And it also means the constructor is not used at all. Sorry for pointing you into the wrong direction.

Reply:
Removed ctor and documented UnmarshalJSON.